packages/zoe/src/zoeService/zoe.js

```diff
@@ -246,8 +250,8 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
       E(adminNode)
         .done()
         .then(
-          () => instanceAdmin.exitAllSeats(),
-          () => instanceAdmin.exitAllSeats(),
+          () => instanceAdmin.shutdown(),
+          () => instanceAdmin.shutdown(),
```

Comment:
On seats, we have `exit` vs `kickOut` to distinguish between exiting for "normal" reasons vs exiting because of a problem. Such problems include circumstances in addition to the contract calling `kickOut`. For example, if an offerHandler's returned promise rejects, the seat is kicked out with the rejection reason, which is good.

IIUC, if the zcf vat crashes, the rejection handler above is called with an error representing the reason for crash. But the resulting shutdown ignores this reason and exits all the seats as if by `exit` rather than as if by `kickOut`. This is a false-successful-exit inconsistency.

Going the other way, on an explicit call to `zcf.shutdown()`, all the seats already created exit as if by `exit` rather than `kickOut`, which seems good. But any offer made after shutdown causes the offer to be rejected. We might view the later as a false-problematic-exit inconsistency. Or at least race condition.

I suggest that we fix the first inconsistency. But I suggest we document rather than "fix" the latter race condition, as I don't see a fix that's actually better than the current behavior.

I note that the one fix I suggest, to the rejection-handler behavior above, is not new as of this PR. Therefore, it is ok if it is not fixed as of this PR.

When we do address these, we may also want to introduce a `shutdown` variant that the contract can explicitly call to say "I'm shutting down abnormally because of the following problem", where that variant would also result in a `kickOut` of all seats.

Reply:
This sounds good to me. The `shutdown` variant you describe should be a good fix for the first issue, in that we can hook it up to the rejection handler.

Reply:
Issue created here: https://github.com/Agoric/agoric-sdk/issues/1774